src/opt/horizontal_fusion_impl.cpp

```diff
@@ -0,0 +1,848 @@
+#include "horizontal_fusion_impl.hpp"
+#include <migraphx/iterator_for.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+static unsigned opcode_bits  = 16;
+static unsigned hash_id_bits = 16;
+static unsigned filter_bits  = 8;
+static unsigned kernel_bits  = 8;
+
+std::unordered_map<std::string, encoder> horizontal_fusion_impl::op_registry =
+    horizontal_fusion_impl::create_op_registery();
+std::unordered_map<std::string, int> horizontal_fusion_impl::op_flag =
+    horizontal_fusion_impl::create_op_flag();
+
+std::unordered_map<std::string, encoder> horizontal_fusion_impl::create_op_registery()
+{
+    std::unordered_map<std::string, encoder> m;
+    m["gpu::convolution"]    = encode_conv_common;
+    m["gpu::conv_bias_relu"] = encode_conv_common;
+    m["hip::add_relu"]       = encode_common;
+    m["convolution"]         = encode_conv_common;
+    m["add"]                 = encode_common;
+    m["relu"]                = encode_common;
+    return m;
+}
+
+std::unordered_map<std::string, int> horizontal_fusion_impl::create_op_flag()
+{
+    std::unordered_map<std::string, int> m;
+    m["gpu::convolution"]    = 1;
```

Comment:
What does these flag means? What does `1` mean? What does `0` mean? What does the other integer values mean as well?

Reply:
I removed this. fixed.